new-lamassu-admin/src/pages/Customers/Wizard.js

```diff
@@ -0,0 +1,137 @@
+import Stepper from 'src/components/Stepper'
+import { Button } from 'src/components/buttons'
+import { comet } from 'src/styling/variables'
+
+import { entryType, customElements } from './helper'
+
+const LAST_STEP = 2
+
+const styles = {
+  stepper: {
+    margin: [[16, 0, 14, 0]]
+  },
+  submit: {
+    display: 'flex',
+    flexDirection: 'row',
+    margin: [['auto', 0, 24]]
+  },
+  button: {
+    marginLeft: 'auto'
+  },
+  form: {
+    height: '100%',
+    display: 'flex',
+    flexDirection: 'column'
+  },
+  infoTitle: {
+    margin: [[18, 0, 20, 0]]
+  },
+  infoCurrentText: {
+    color: comet
+  },
+  blankSpace: {
+    padding: [[0, 30]],
+    margin: [[0, 4, 0, 2]],
+    borderBottom: `1px solid ${comet}`,
+    display: 'inline-block'
+  }
+}
+
+const useStyles = makeStyles(styles)
+
+const getStep = (step, selectedValues) => {
+  switch (step) {
+    case 1:
+      return entryType
+    case 2:
+      return customElements[selectedValues?.dataType]
+    default:
+      return Fragment
+  }
+}
+
+const GetValues = ({ setValues }) => {
+  const { values } = useFormikContext()
+
+  useEffect(() => {
+    setValues && values && setValues(values)
+  }, [setValues, values])
+
+  return null
+}
+
+const Wizard = ({ onClose, save, error }) => {
+  const classes = useStyles()
+
+  const [selectedValues, setSelectedValues] = useState(null)
+  const [liveValues, setLiveValues] = useState({})
+
+  const [{ step, config }, setState] = useState({
+    step: 1
+  })
+
+  // TODO forms error handling
+  const isLastStep = step === LAST_STEP
+  const stepOptions = getStep(step, selectedValues, liveValues)
+
+  const onContinue = async it => {
+    const newConfig = R.merge(config, stepOptions.schema.cast(it))
+    setSelectedValues(newConfig)
+
+    if (isLastStep) {
+      return save(newConfig)
+    }
+
+    setState({
+      step: step + 1,
+      config: newConfig
+    })
+  }
+
+  return (
+    <>
+      <Modal
+        title="Manual data entry"
+        handleClose={onClose}
+        width={520}
+        height={520}
+        open={true}>
+        <Stepper
+          className={classes.stepper}
+          steps={LAST_STEP}
+          currentStep={step}
+        />
+        <Formik
+          validateOnBlur={false}
+          validateOnChange={false}
+          enableReinitialize
+          onSubmit={onContinue}
+          initialValues={stepOptions.initialValues}
+          validationSchema={stepOptions.schema}>
+          <Form className={classes.form}>
+            <GetValues setValues={setLiveValues} />
+            <stepOptions.Component
+              selectedValues={selectedValues}
+              {...stepOptions.props}
+            />
+            <div className={classes.submit}>
+              {error && <ErrorMessage>Failed to save</ErrorMessage>}
+              <Button className={classes.button} type="submit">
+                {isLastStep ? 'Add Data' : 'Next'}
+              </Button>
+            </div>
+          </Form>
```

Comment:
You can have access to the formik values without having to create a useEffect in a component for only that purpose. I have the feeling I've already seen something similar before, but not sure where. Either way, the Formik node gives you access to the props in this way: https://formik.org/docs/api/formik#example

```suggestion
          {({ values }) => (
            <Form className={classes.form}>
              <GetValues setValues={setLiveValues} />
              <stepOptions.Component
                selectedValues={selectedValues}
                {...stepOptions.props}
              />
              <div className={classes.submit}>
                {error && <ErrorMessage>Failed to save</ErrorMessage>}
                <Button className={classes.button} type="submit">
                  {isLastStep ? 'Add Data' : 'Next'}
                </Button>
              </div>
            </Form>
            )
          }
```

Reply:
right, this is fixed on the base PR already, should be ok after rebase